framework/console/controllers/BaseMigrateController.php

```diff
@@ -877,6 +900,13 @@ protected function generateMigrationSourceCode($params)
         return $this->renderFile(Yii::getAlias($this->templateFile), $params);
     }
 
+
+    /**
+     * Truncates the database and reapplies all migrations from the beginning.
+     * @return void
+     */
+    abstract protected function refreshDatabase();
```

Comment:
abstract function will introduce BC break, extensions that do not implement it, will break. It should be implemented and just print a message that it is not implemented.